library/Mockery/Generator/Parameter.php

```diff
@@ -48,7 +48,13 @@ public function getTypeHintAsString()
         if ((version_compare(PHP_VERSION, '5.4.1') >= 0)) {
             try {
                 if ($this->rfp->getClass()) {
-                    return $this->rfp->getClass()->getName();
+                    $typehintedClass = $this->rfp->getClass()->getName();
+
+                    if ($this->isNamespacedClass($typehintedClass)) {
```

Comment:
I was thinking the same. Reflection always gets the FQN of the typehinted class.